check/check_db_dump_recent.py

```diff
@@ -43,17 +45,30 @@ def run(self, log_file_path):
 
                 return RETURN_CODE_CRITICAL
 
-            # Convert the last line of the log to a timestamp.
-            # Use a inner try...except block to handle the somewhat more
-            # expected error of "the failover process ran, but something
-            # went wrong" differently.
-            try:
-                last_success = datetime.strptime(
-                    last_line,
-                    "%Y-%m-%dT%H:%M:%S%z",
+            # Assume the failover process has never run, then attempt to
+            # disprove that by looping through the logs.
+            last_success = None
+            for line in reversed(line_list):
+                # If OK_STRING is in the line we are looking at, we need to extract
+                # the timestamp from that line to determine when the failover
+                # process last succeeded.
+                if self.OK_STRING in line:
+                    last_success_timestamp = line.split(" ")[0]
+                    last_success = datetime.strptime(
+                        last_success_timestamp,
+                        "%Y-%m-%dT%H:%M:%S+0000",
+                    )
+                    # We only want the most recent success, so once we have
+                    # found it, break out of this loop
+                    break
+
+            if last_success is None:
+                print(
+                    "The failover process has never succeeded, "
+                    "according to the %s." % (
```

Comment:
```suggestion
                    "according to %s." % (
```

The definite article sounds a bit weird before a file path.